src/main.rs

```diff
@@ -43,6 +44,57 @@ struct Args {
     command: Vec<String>,
 }
 
+// TODO: Remove once `OnceCell::get_or_try_init()` is stable.
+pub(crate) trait OnceCellExt<T> {
+    fn get_or_try_init_<F, E>(&self, f: F) -> Result<&T, E>
+    where
+        F: FnOnce() -> Result<T, E>;
+}
+
+impl<T> OnceCellExt<T> for OnceCell<T> {
+    fn get_or_try_init_<F, E>(&self, f: F) -> Result<&T, E>
+    where
+        F: FnOnce() -> Result<T, E>,
+    {
+        if let Some(value) = self.get() {
+            Ok(value)
+        } else {
+            let value = f()?;
+            Ok(self.get_or_init(|| value))
+        }
+    }
+}
+
+/// A type representing a log that creates the associated file lazily
+/// upon first write.
+#[derive(Default)]
+struct DeferredLog {
+    file: OnceCell<File>,
+}
+
+impl DeferredLog {
+    fn file(&mut self) -> io::Result<&File> {
+        self.file.get_or_try_init_(|| {
+            fs::OpenOptions::new()
+                .create(true)
+                .append(true)
+                .open(".vmtest.log")
+        })
+    }
```

Comment:
What happens if the file fails to create? Does `log` print a warning and then continue along? 

Maybe it's better/simpler to `unwrap()` the file creation. At least that way you wouldn't need to have the try init dance

Reply:
It would be bubbled up through the `write` / `flush` call. We can also unwrap. I don't have a strong preference.

Reply:
Right, but in our code which just does `info!(...)` we are not checking for errors or anything. If that happens to just panic, we may as well panic during file creation here.

Reply:
I checked and it does not panic but just swallow any errors.

Reply:
Thanks for checking. In that case we'd probably want to panic - silently dropping logs is not very good

Reply:
Sure thing. Changed.